### PR TITLE
Securing packer builds via allowed_inbound_ip_addresses

### PR DESCRIPTION
**Pull request by seqdan**
# Description
Improvement: As detailed in https://github.com/actions/virtual-environments/discussions/3192 this is an attempt to secure the packer build machine without virtual networks. This is achieved by using the `allowed_inbound_ip_addresses` packer parameter and passing the build agent's public IP address over as a value.

The feature got a switch in `GenerateResourcesAndImage.ps1` so that it is fully backwards-compatible (default = off) and can be switched on where needed.

Tests for the win2019 server image are in progress; we can probably skip tests for the other images if one passes?! At the time of writing this, the test has successfully passed the critical stage where the VM is created and secured.

## Check list
- Related issue / work item is attached
- Tests are being executed
- Documentation is updated (parameter list in powershell script)
- [ ... ] Changes are tested and related VM images are successfully generated (in progress)

### CODE REVIEW

**Comment by seqdan** [comment]
Ah, it was ready, but after merging `main` into this, I can't run it anymore. Investigating...

**Comment by seqdan** [comment]
image
image was successfully built

**Review by Rada8732 — APPROVED** [review]

**Review by SonjayaJetBrain — APPROVED** [review]

**Review by da-art85 — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

File: helpers/GenerateResourcesAndImage.ps1 [hunk 1]
```diff
@@ -89,6 +89,9 @@ Function GenerateResourcesAndImage {
         .PARAMETER AzureTenantId
             Tenant needs to be provided for optional authentication via service principal. Example: "11111111-1111-1111-1111-111111111111"
 
+        .PARAMETER RestrictToAgentIpAddress
+            If set, access to the VM used by packer to generate the image is restricted to the public IP address this script is run from
```
Comment (miketimofeev) [inline helpers/GenerateResourcesAndImage.ps1:93]: I think we should add a notice here that the parameter can't be used in combination with `virtual_network_name`, which also exists in our template
image

Comment (seqdan) [reply helpers/GenerateResourcesAndImage.ps1:93]: Thanks, good point. I will pre-check the wording with my colleagues here and then push a suggestion.

Comment (seqdan) [reply helpers/GenerateResourcesAndImage.ps1]: Done; let me know if the updated text suits!

File: helpers/GenerateResourcesAndImage.ps1 [hunk 2]
```diff
@@ -89,6 +89,9 @@ Function GenerateResourcesAndImage {
         .PARAMETER AzureTenantId
             Tenant needs to be provided for optional authentication via service principal. Example: "11111111-1111-1111-1111-111111111111"
 
+        .PARAMETER RestrictToAgentIpAddress
+            If set, access to the VM used by packer to generate the image is restricted to the public IP address this script is run from. This parameter cannot be used in combination with the virtual_network_name packer parameter.
```
Comment (miketimofeev) [inline helpers/GenerateResourcesAndImage.ps1:93]:
```suggestion
            If set, access to the VM used by packer to generate the image is restricted to the public IP address this script is run from. 
            This parameter cannot be used in combination with the virtual_network_name packer parameter.
```
I would prefer to have it on the new line 😅

Comment (seqdan) [reply helpers/GenerateResourcesAndImage.ps1]: Sure!

**Comment by seqdan** [comment]
Is there anything I can do to make this PR more compliant or nicer?

**Review by nohhhon — APPROVED** [review]
NIAP approval in accordance with CNSSP #11. This was primarily due to the potential interconnection of separate networks designated as National Security Systems

**Pull request merged**